cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -101,8 +95,12 @@ public Boolean call() throws Exception {
         if (!getMonitorClient().allSystemServicesOk()) {
           return false;
         }
-        // check that the dataset service is up
-        getNamespaceClient().list();
+        // check that the dataset service is up, and also that the default namespace exists
+        try {
+          getNamespaceClient().get(Id.Namespace.DEFAULT);
```

Comment:
Sure, but either way, this logic will be needed to ensure that the Default namespace is available before starting tests. The MonitorClient#allServicesOk will not guarantee that.